.github/workflows/README.md

```diff
@@ -0,0 +1,58 @@
+## CI/CD
+
+There are 5 main stages that run on GitHub actions:
+
+* Linting
+* Test
+* Test `main` versions
+* Package
+* Release
+
+There are some other stages that run for every push on the main branches:
+
+* [Microbenchmark](./microbenchmark.yml)
+* [Snapshoty](./snapshoty.yml)
+
+### Scenarios
+
+* Compatibility matrix runs on branches, tags and PRs basis.
+* Tests should be triggered on branch, tag and PR basis.
+* Commits that are only affecting the docs files should not trigger any test or similar stages that are not required.
+* Automated release in the CI gets triggered when a tag release is created.
+* **This is not the case yet**, but if Github secrets are required then Pull Requests from forked repositories won't run any build accessing those secrets. If needed, then create a feature branch.
+
+### Compatibility matrix
+
+Ruby agent supports compatibility to different ruby versions and frameworks, those are defined in:
+
+* [frameworks](https://github.com/elastic/apm-agent-ruby/blob/main/.ci/framework.yml) for all the PRs.
+* [frameworks](https://github.com/elastic/apm-agent-ruby/blob/main/.ci/.main_framework.yml) for all the PRs using the `main` branches.
+* Ruby [versions](https://github.com/elastic/apm-agent-ruby/blob/main/.ci/.ruby.yml) for all the `*nix` builds.
+* [Exclude list](https://github.com/elastic/apm-agent-ruby/blob/main/.ci/exclude.yml) for the above entries.
```

Comment:
We can use relative paths here.

> GitHub will automatically transform your relative link or image path based on whatever branch you're currently on, so that the link or path always works.

Source: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes#relative-links-and-image-paths-in-readme-files

That way, the link will be valid if you open a different branch or tag. (In case there are changes)